src/structures/VoiceChannel.js

```diff
@@ -103,6 +103,7 @@ class VoiceChannel extends GuildChannel {
    *   .catch(console.error);
    */
   setUserLimit(userLimit, reason) {
+    if (typeof userLimit !== 'number' || userLimit === null) return Promise.reject(new TypeError('USERLIMIT_INVALID_TYPE', 'userLimit', 'an intenger')); // eslint-disable-line max-len
```

Comment:
Discord seems to validate that you actually send them a number on their end, so no need to do so on our end as well.
`null` however does indeed do nothing, and could be rejected or handled like 0 as someone might excpect it to work. (For example: Passing `ClientUser#setAvatar` null, resets the avatar)
I am fine with either, should just be clarified.

Reply:
well i only tried null and undefined though it would be the same for every other type but saw the line 
```js
user_limit: data.userLimit != null ? data.userLimit : this.userLimit, // eslint-disable-line eqeqeq
```
in GuildChannel too late, but i think i will adapt the idea of handle null like 0 here

Reply:
I think it would be better if you change the line in `GuildChannel#edit` to only check for `!== undefined` instead of `!= null`.
That way the edit method would also accept `null` to reset the user limit.

Reply:
okay i will edit that quick